### ISSUE TITLE
Provide more meaningful error messages when the `kit push` command fails

### ISSUE THREAD

**Issue by brett-hodges** [jozu-ai/kitops#394]
**Describe the problem you're trying to solve**
`kit push` returns an ambiguous error message when the command fails because a Registry doesn't support Repository names with multiple-levels beyond what is expected.

For example, the current error message use case mentioned above is:

`Pushing jozu.ml/jozu-test/repo-level-testing/mymodel:latest`
`Failed to push: failed to copy to remote: Not Found (additional info: {"status":404,"error":"Not Found"})`

**Describe the solution you'd like**
The error message should clearly specify that the issue is with multi-level Repository name.
For example, consider:
`Failed to push: failed to copy to remote registry: Multi-level repository names are not supported at this time.`

**Comment by Shawn-Dong** [comment]
Hi, I've been looking into this issue and have a rough idea of how to approach it. I'd love to work on a fix. Could you assign this to me?

**Comment by brett-hodges** [comment]
We generally only assign issues to maintainers. Please reply back with a
comment detailing how you'd code and test this issue and that will help
others know you're working on it.

**Comment by Shawn-Dong** [comment]
> We generally only assign issues to maintainers. Please reply back with a
> comment detailing how you'd code and test this issue and that will help
> others know you're working on it.
> […](#)

 Hi! Here's my plan for this:
                                                                                                                                                                                                                                                                             
Approach:` In pkg/cmd/push/cmd.go`, the runCommand function already handles some HTTP error codes (e.g. 401). I'd add a check for 404 errors where the destination repository contains multiple path segments (e.g. org/sub/repo), and display a clearer message pointing to multi-level repo names as the likely cause.

One thing worth noting: the error in the issue hits the generic error branch, not the errcode.ErrorResponse branch, because the registry returns a non-standard error body. So the fix needs to cover both paths.

Testing: I'd add unit tests covering:
  - 404 with a multi-level repo name → hint message shown
  - 404 with a single-level repo name → no hint (avoid false positives)
  - Non-structured error containing "not found" with a multi-level repo → hint shown

  Happy to take this on!

**Comment by amisevsk** [comment]
I don't think that catches the issue fully -- we'd be showing that message when it does apply, but there's no way in the OCI spec to distinguish between other reasons for returning a 404. The registry might return a 404 because it doesn't understand the API path (technically a bug, if you're adhering to the OCI specification) but it might return a 404 for other reasons potentially. 

I'm not sure there's a straightforward way to solve this issue that doesn't produce false positives. Additionally, hitting this bug generally means deviating from expectations on how OCI references generally work, so I think this issue should instead just be closed as "known, won't fix".
